packages/carbon-graphs/src/js/controls/Graph/helpers/helpers.js

```diff
@@ -264,6 +264,66 @@ const getAxesLabelCharacterLimits = (config) => ({
   xAxisLimit: getXAxisWidth(config) / 7.33,
   yAndY2AxisLimit: config.height / 8,
 });
+/**
+ * Adds onClick event for each label in Axes.
+ * Criteria:
+ *  * Text needs to have a valid function for onClick
+ *  * Label should exceed the calculated character limit with respect to axis length
+ *
+ * @private
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @param {string} className - class name of axis
+ * @param {object} axisObj - axis object provided by the consumer
+ * @param {number} charLimit - character limit of label in axis
+ * @param {string} axisType - type of axis
+ * @returns {object} d3 svg path
+ */
+const addLabelEventHandler = (
+  canvasSVG,
+  className,
+  axisObj,
+  charLimit,
+  axisType,
+) => canvasSVG.selectAll(`.${className} text`).each(function () {
+  // eslint-disable-next-line no-unused-expressions
+  shouldTruncateLabel(axisObj.label, charLimit)
+    ? d3
+      .select(this)
+      .style('cursor', 'pointer')
+      .on('click', () => {
+        // eslint-disable-next-line no-unused-expressions
+        utils.isDefined(axisObj.onLabelClick)
+            && utils.isFunction(axisObj.onLabelClick)
+          ? axisObj.onLabelClick(axisObj.label, d3.select(this))
+          : loadLabelPopup(axisObj.label, axisType);
+      })
+    : (null);
+});
+
+/**
+ * Removes label click event if label is small during reflow.
+ *
+ * @param {object} config - config object derived from input JSON
+ * @returns {undefined} - returns nothing
+ */
+const handleLabelClickFunctionDuringReflow = (config) => {
+  const axesLabelCharLimits = getAxesLabelCharacterLimits(config);
+  const selectLabelTextElement = (className) => document.querySelector(`.${className} text`);
+  const setClickFunction = (className) => d3.select(selectLabelTextElement(className)).on('click', null).style('cursor', 'auto');
+
+  if (config.axis.x.label && !shouldTruncateLabel(config.axis.x.label, axesLabelCharLimits.xAxisLimit)
+      && ((selectLabelTextElement(styles.axisLabelX) === null) ? false : (utils.isFunction(d3.select(selectLabelTextElement(styles.axisLabelX)).on('click'))))) {
```

Comment:
This would be easier to read if you broke up the segments of the check into their own verbose variables:
```jsx
const shouldTruncateXLabel = config.axis.x.label && shouldTruncateLabel(config.axis.x.label, axesLabelCharLimits.xAxisLimit);
const xLabelHasOnClick = (selectLabelTextElement(styles.axisLabelX) === null) ? false : utils.isFunction(d3.select(selectLabelTextElement(styles.axisLabelX)).on('click')));
if (!shouldTruncateXLabel && xLabelHasOnClick) {
    setClickFunction(styles.axisLabelX);
}
```

Might even get away with another helper to dry this up?

```jsx
const handleLabelClickFunctionDuringReflow = (config) => {
  const axesLabelCharLimits = getAxesLabelCharacterLimits(config);
  const selectLabelTextElement = (className) => document.querySelector(`.${className} text`);  
  const updateOnClickForAxis = (axisLabel, axisLimit, axisStyles) => {
    const shouldTruncateLabel = axisLabel && shouldTruncateLabel(axisLabel, axisLimit);
    const labelHasOnClick = (selectLabelTextElement(axisStyles) === null) ? false : utils.isFunction(d3.select(selectLabelTextElement(axisStyles)).on('click')));
    if (!shouldTruncateLabel && labelHasOnClick) {
      d3.select(selectLabelTextElement(axisStyles)).on('click', null).style('cursor', 'auto');
    }
  }

  updateOnClickForAxis(config.axis.x.label, axesLabelCharLimits.xAxisLimit, styles.axisLabelX);
  updateOnClickForAxis(config.axis.y.label, axesLabelCharLimits.yAndY2AxisLimit, styles.axisLabelY);
  updateOnClickForAxis(config.axis.y2.label, axesLabelCharLimits.yAndY2AxisLimit, styles.axisLabelY2);
}

Reply:
Made changes [here](https://github.com/cerner/terra-graphs/pull/239/commits/bfe931955c01cd8c2fc95f2945a72cde4821d6eb).